panic: GW panic when stopping and dashboard not available

**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

Assuming the dashboard is stopped. We attempt to stop the gateway. GW attempts to de-register with the dashboard, and panics.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17e268b]

goroutine 1 [running]:
main.(*HTTPDashboardHandler).DeRegister(0xc4203bd4f0, 0x0, 0x0)
	/Users/ahmet/go/src/github.com/TykTechnologies/tyk/dashboard_register.go:188 +0x17b
main.main()
	/Users/ahmet/go/src/github.com/TykTechnologies/tyk/main.go:1031 +0x90b
```

**What is the expected behavior?**

Should not panic

--- a/dashboard_register.go
+++ b/dashboard_register.go
@@ -184,11 +184,15 @@ func (h *HTTPDashboardHandler) DeRegister() error {
 
 	c := &http.Client{Timeout: 5 * time.Second}
 	resp, err := c.Do(req)
-	if err != nil || resp.StatusCode != 200 {
-		return fmt.Errorf("request failed with code %d and error %v", resp.StatusCode, err)
+
+	if err != nil {
+		return fmt.Errorf("deregister request failed with error %v", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("deregister request failed with status %v", resp.StatusCode)
 	}
 
-	defer resp.Body.Close()
 	val := NodeResponseOK{}
 	if err := json.NewDecoder(resp.Body).Decode(&val); err != nil {
 		return err

